### ISSUE TITLE
The problem of transformer learning rate adjustment.

### ISSUE THREAD

**Issue by 7zhanghao** [espnet/espnet#4286]
Thanks to the implementation of various tasks in egs, we quickly get a good baseline on this basis. 
However, due to differences in GPU types and memory, as well as some other structural changes, we usually need to adjust the learning rate by ourselfs.
Taking the transformer as an example, how to adjust the scaling factor and the number of warmup steps. In recent experiments, I always seem to be trying aimlessly.
Hope you can give some useful suggestions, because most of the baselines in egs achieve better results than other frameworks implementations, and I'd really like to know how you tune the learning rate to achieve this.

**Comment by sw005320** [comment]
Thanks for raizing the issue.
I usually try the following steps:

1. pick up a recipe and the corresponding config file that is close to your target task
2. check the total batch size (= batch size (`batch_bins` or `batch_size`) x `accum_grad`). See https://espnet.github.io/espnet/espnet2_training_option.html#change-mini-batch-type and https://espnet.github.io/espnet/espnet2_training_option.html#gradient-accumulating
3. adjust the batch size and `accum_grad` to fit it to your GPU memory and maintain the same range as the above total batch size
4. tuning the learning rate (You can just check the first several epochs of the learning curve, and if you find some degradation, you can reduce the learning rate).

In my experience, most recipes work well with this way of tuning.
Suppose your data is very small, noisy (noisy audio, containing many transcription errors), etc. In that case, we need to tune the other parameter (total batch size, warmup steps, feature extraction configurations, model architecture, disable data augmentation, etc.).
The other way is to use pre-trained models, e.g., hubert.

**Comment by 7zhanghao** [comment]
> Thanks for raizing the issue. I usually try the following steps:
> 
> 1. pick up a recipe and the corresponding config file that is close to your target task
> 2. check the total batch size (= batch size (`batch_bins` or `batch_size`) x `accum_grad`). See https://espnet.github.io/espnet/espnet2_training_option.html#change-mini-batch-type and https://espnet.github.io/espnet/espnet2_training_option.html#gradient-accumulating
> 3. adjust the batch size and `accum_grad` to fit it to your GPU memory and maintain the same range as the above total batch size
> 4. tuning the learning rate (You can just check the first several epochs of the learning curve, and if you find some degradation, you can reduce the learning rate).
> 
> In my experience, most recipes work well with this way of tuning. Suppose your data is very small, noisy (noisy audio, containing many transcription errors), etc. In that case, we need to tune the other parameter (total batch size, warmup steps, feature extraction configurations, model architecture, disable data augmentation, etc.). The other way is to use pre-trained models, e.g., hubert.

Thanks for your kind advice.
